### PR TITLE
feat: render library into `Sidebar` on mobile

### PR DESCRIPTION
**Pull request by dwelle**
- add `ref` to `<Sidebar>`
- consolidate library rendering by rendering the same component in both breakpoints
- refactor, split components

Should have more or less behavior parity to production except the Sidebar on mobile thing of course. Haven't optimized perf, that we'll need to do separately.

### CODE REVIEW

File: src/components/LibraryMenu.tsx [hunk 1]
```diff
@@ -162,139 +257,50 @@ export const LibraryMenu = ({
...
+        trackEvent(
+          "library",
+          `toggleLibraryDock (${docked ? "dock" : "undock"})`,
+          `sidebar (${device.isMobile ? "mobile" : "desktop"})`,
+        );
+      }}
+      ref={ref}
+    >
+      <Sidebar.Header className="layer-ui__library-header">
+        <LibraryMenuHeaderContent
```
Comment (ad1992) [inline src/components/LibraryMenu.tsx:278]: let's rename it to `LibraryMenuHeader` instead as that conveys its the header

File: src/components/LibraryMenuHeaderContent.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+      {publishLibSuccess && renderPublishSuccess()}
+      {!itemsSelected && (
+        <ToolButton
+          key="import"
+          type="button"
+          title={t("buttons.load")}
+          aria-label={t("buttons.load")}
+          icon={load}
+          onClick={async () => {
+            try {
```
Comment (ad1992) [inline src/components/LibraryMenuHeaderContent.tsx:166]: can be moved to a function

File: src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -19,103 +25,115 @@ import { updateObject } from "../../utils";
...
-        setHostSidebarCounters((s) => ({
-          rendered: s.rendered - 1,
-          docked: isDockedFallback ? s.docked - 1 : s.docked,
-        }));
-      };
-    }
-  }, [__isInternal, setHostSidebarCounters, isDockedFallback]);
+      useLayoutEffect(() => {
+        if (docked === undefined) {
+          // ugly hack to get initial state out of AppState without susbcribing
```
Comment (ad1992) [inline src/components/Sidebar/Sidebar.tsx:59]:
```suggestion
          // ugly hack to get initial state out of AppState without subscribing
```
typo

File: src/packages/excalidraw/README.md [hunk 4]
```diff
@@ -643,6 +643,7 @@ The `<Sidebar>` component takes these props (all are optional except `children`)
 | onDock | <pre>(isDocked: boolean) => void</pre> | Invoked when the user toggles the dock button. |
 | docked | boolean | Indicates whether the sidebar is docked. By default, the sidebar is undocked. If passed, the docking becomes controlled, and you are responsible for updating the `docked` state by listening on `onDock` callback. See [here](#dockedSidebarBreakpoint) for more info docking. |
 | dockable | boolean | Indicates whether the sidebar can be docked by user (=the dock button is shown). If `false`, you can still dock programmatically by passing `docked=true` |
+| ref | <pre>React.Ref&lt;HTMLDivElement&gt;</pre> | forward ref |
```
Comment (ad1992) [inline src/packages/excalidraw/README.md:646]: Since this going to be removed in future PR, lets not add it here

Comment (dwelle) [reply src/packages/excalidraw/README.md:646]: it's likely not gonna get removed. Refs in exported components are a good idea in general. It's just we may not use it internally.

**Review by ad1992 — APPROVED** [review]

**Pull request merged**